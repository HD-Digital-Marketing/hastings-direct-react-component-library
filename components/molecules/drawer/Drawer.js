import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Icon from '../../atoms/icon/Icon'
import classNames from 'classnames'
import './drawer.scss'
import CloseIcon from '../../../src/img/icons/cross.svg'

const Drawer = (props) => {

    const {
        isOpen = false,
        toggleDrawer,
        className,
        children
    } = props

    const drawerClass = classNames('m-drawer__wrap', className, {
        'm-drawer__wrap--show': isOpen
    })

    return ReactDOM.createPortal(
        <>
            <div className={drawerClass}>
                <div className="m-drawer__background" onClick={toggleDrawer} />
                <div className="m-drawer">
                    <div className="m-drawer__container">
                        <div className="m-drawer__header">
                            <div>
                                <Icon
                                    svgImage={CloseIcon}
                                    svgWidth={18}
                                    svgHeight={18}
                                    isRound={true}
                                    svgFill="#011831"
                                    svgBorderColor="#60759e"
                                    className="m-drawer__icon"
                                    data-dismiss="modal"
                                    aria-label="Close"
                                    onClick={toggleDrawer} />
                            </div>
                        </div>
                        <div className="m-drawer__content">
                            {children}
                        </div>
                    </div>
                </div>
            </div>
        </>, document.body
    )
}

Drawer.propTypes = {
    isOpen: PropTypes.bool,
    toggleDrawer: PropTypes.func,
    className: PropTypes.string,
    children: PropTypes.node
}

export default Drawer