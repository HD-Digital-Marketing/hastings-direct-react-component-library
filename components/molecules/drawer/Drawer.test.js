import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Drawer from './Drawer'

describe('Check the drawer renders', () => {

    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element
        })
    })

    afterEach(() => {
        ReactDOM.createPortal.mockClear()
    })

    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Drawer isOpen={true}>Overlay content</Drawer>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})