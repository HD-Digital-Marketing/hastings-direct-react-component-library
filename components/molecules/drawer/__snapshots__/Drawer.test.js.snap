// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the drawer renders it should match the snapshot 1`] = `
<div
  className="m-drawer__wrap m-drawer__wrap--show"
>
  <div
    className="m-drawer__background"
  />
  <div
    className="m-drawer"
  >
    <div
      className="m-drawer__container"
    >
      <div
        className="m-drawer__header"
      >
        <div>
          <span
            className="a-icon m-drawer__icon a-icon--rounded"
            style={
              Object {
                "backgroundColor": undefined,
                "borderColor": "#60759e",
                "borderWidth": 1.7999999999999998,
                "height": 36,
                "width": 36,
              }
            }
          />
        </div>
      </div>
      <div
        className="m-drawer__content"
      >
        Overlay content
      </div>
    </div>
  </div>
</div>
`;
